internal/sign/job/signer.go

```diff
@@ -144,3 +185,43 @@ func (m *signer) MakeJobTemplate(
 
 	return job, nil
 }
+
+func (s *signer) getHashAnnotationValue(ctx context.Context, privateSecret, publicSecret, namespace string, podTemplate *v1.PodTemplateSpec) (uint64, error) {
+	privateKeyData, err := s.getSecretData(ctx, privateSecret, constants.PrivateSignDataKey, namespace)
+	if err != nil {
+		return 0, fmt.Errorf("failed to get private secret %s for signing: %v", privateSecret, err)
+	}
+	publicKeyData, err := s.getSecretData(ctx, publicSecret, constants.PublicSignDataKey, namespace)
+	if err != nil {
+		return 0, fmt.Errorf("failed to get public secret %s for signing: %v", publicSecret, err)
+	}
+
+	return getHashValue(podTemplate, publicKeyData, privateKeyData)
+}
```

Comment:
I see that build is doing something similar.
Should we extract it to a `Hash` API (with different implementation for `build` and `sign`) and be consumed in `build` and `sign` logic?

Same for `getHashValue`

Reply:
it means that you want to define them as an interface, right? but what's the point? it is not a problem to test it in unit test. We will end up with 2 different implementation  ( just as we have right now), but it addition the API will not be as clear as it is now

Reply:
The point is that this is not a build/sign logic, it is a hash logic, extracted to it's own API and consumed by another API (using dependency injection)

Reply:
As agreed, `Hash` will be extracted to its own API and consumed twice in `build` and `sign`.
https://github.com/kubernetes-sigs/kernel-module-management/issues/184.